CorsixTH/Lua/hospital.lua

```diff
@@ -256,6 +249,18 @@ function Hospital:Hospital(world, avail_rooms, name)
   end
 end
 
+--! Checks if a room has been discovered
+--!param room_id (string) The name of the room
+--!return (boolean) true if found
+function Hospital:checkRoomDiscovery(room_id)
+  for _, room_idx in ipairs(self.room_discovery) do
```

Comment:
The `_idx` suffix implies 'index', but the second returned value from `ipairs` is an element from the `self.room_discovery` array, which is not an index, if I may believe `world:getAvailableRooms()`.